src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -236,16 +237,16 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			} else {
 				let result = undefined;
 				const isFile = URI.parse(linkCalloutResult?.insertUnescapedLinkUrl).scheme === 'file';
-				if (!path.isAbsolute(linkCalloutResult?.insertUnescapedLinkUrl) && isFile) {
+				if (isFile && !path.isAbsolute(linkCalloutResult?.insertUnescapedLinkUrl)) {
 					const notebookDirName = path.dirname(this.cellModel?.notebookModel?.notebookUri.fsPath);
-					const relativePath = linkCalloutResult?.insertUnescapedLinkUrl.replace(/(\.\\ | \.\/)/g, '/');
+					const relativePath = (linkCalloutResult?.insertUnescapedLinkUrl).replace(/\\/g, path.posix.sep);
 					result = path.resolve(notebookDirName, relativePath);
 				}
 				// Otherwise, re-focus on the output element, and insert the link directly.
 				this.output?.nativeElement?.focus();
 				result = result ?? linkCalloutResult?.insertUnescapedLinkUrl;
-				// Callout is responsible for returning escaped strings
-				document.execCommand('insertHTML', false, `<a href="${escape(result)}">${escape(linkCalloutResult?.insertUnescapedLinkLabel).replace(/%20/g, ' ')}</a>`);
+				// Callout is responsible for returning escaped strings. We replace %20 for whitespaces to allow spaces in the link's display text
+				document.execCommand('insertHTML', false, `<a href="${encodeURI(result)}">${escape(linkCalloutResult?.insertUnescapedLinkLabel).replace(/%20/g, ' ')}</a>`);
```

Comment:
For these I think you should just use https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/common/strings.ts#L40 instead - this isn't a URI and so we shouldn't be using a function to encode URIs here

Reply:
Let me try using that instead